### PR TITLE
Adapt test case finder API

### LINKED ISSUE #205: Parameterized tests are overwritten
Suppose the following recheck-web test:

```java
@ParameterizedTest
@MethodSource( "com.github.beatngu13.visualtesting.util.DriverFactory#getAll" )
void test(WebDriver driver) throws Exception {
	re.startTest();
	driver.get("https://example.com/");
	re.check(driver, "example");
	re.capTest();
}
```

If the user doesn't provide a test name via `startTest(String)` and uses the default `startTest()`, we simply overwrite the previous GM as all parameterized test cases have the same method name.

We should try to recognize this and then at least append a suffix.

### PR DESCRIPTION
**Pull request by beatngu13**
Two reviewers on purpose, please don't approve prematurely.

Resolves #205.

### CODE REVIEW

**Review by varlog23 — COMMENTED** [review]
Everything works as expected. Could not find any flaws with this approach.

File: src/main/java/de/retest/recheck/TestCaseFinder.java [hunk 1]
```diff
@@ -32,51 +37,128 @@ private TestCaseFinder() {}
...
+	 * @return Test case information for the test case method in the given stack trace.
+	 */
+	public static TestCaseInformation findTestCaseMethodInStack( final StackTraceElement[] trace ) {
+		for ( final StackTraceElement element : trace ) {
+			final TestCaseAnnotationType type = determineTestCaseAnnotationType( element );
+			if ( type == TestCaseAnnotationType.NORMAL ) {
+				return new TestCaseInformation( element, type, 1 );
+			}
+			if ( type == TestCaseAnnotationType.REPEATABLE ) {
+				final int count = repeatableTestCaseAnnotationsCount.merge( element, 1, Math::addExact );
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/TestCaseFinder.java]: Use `Integer#sum( int, int )`.

File: src/main/java/de/retest/recheck/TestCaseFinder.java [hunk 2]
```diff
@@ -32,51 +37,128 @@ private TestCaseFinder() {}
...
 		}
 
 		final Annotation[] annotations = method.getAnnotations();
 		for ( final Annotation annotation : annotations ) {
 			final String annotationName = annotation.annotationType().getName();
 			if ( testCaseAnnotations.contains( annotationName ) ) {
-				return true;
+				return TestCaseAnnotationType.NORMAL;
+			}
+			if ( repeatableTestCaseAnnotations.contains( annotationName ) ) {
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/TestCaseFinder.java:165]: Change order.

File: src/main/java/de/retest/recheck/TestCaseFinder.java [hunk 3]
```diff
@@ -32,51 +37,128 @@ private TestCaseFinder() {}
...
+	private static final Set<String> repeatableTestCaseAnnotations = new HashSet<>( Arrays.asList( //
+			// JUnit Vintage (v4)
+			"org.junit.experimental.theories.Theory", //
+			// JUnit Jupiter (v5)
+			"org.junit.jupiter.api.RepeatedTest", //
+			"org.junit.jupiter.params.ParameterizedTest" ) );
+
+	private static final Map<StackTraceElement, Integer> repeatableTestCaseAnnotationsCount = new HashMap<>();
+
+	private TestCaseFinder() {}
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/TestCaseFinder.java]: To singleton.

File: src/main/java/de/retest/recheck/TestCaseFinder.java (hunk 2 above)
Comment (beatngu13) [reply src/main/java/de/retest/recheck/TestCaseFinder.java:165]: I think I'll keep it as it is, natural order. Having two annotations is just wrong usage I guess.

**Review by diba1013 — APPROVED** [review]

**Pull request merged**